src/middlewares/methodWhitelist.ts

```diff
@@ -0,0 +1,12 @@
+import { Request, Response, NextFunction } from 'express';
+import { methods } from '../api';
+
+const allowedMethods = Object.keys(methods);
+
+export const methodWhitelist = (req: Request, res: Response, next: NextFunction) => {
+  const method = req.body?.method;
```

Comment:
If there is no method we are falling back to the next(). Maybe we can invert the check to call next and if not fail with 403.

Reply:
original goal - if method is in whitelist we continue with next, if not, return error, correct ?